common/s3util.cpp

```diff
@@ -214,6 +217,59 @@ SdkGetObjectResponse S3Util::sdkGetObject(const string& key,
   return getObjectResult;
 }
 
+
+void S3Util::listObjectsV2Helper(const string& prefix, const string& delimiter,
```

Comment:
`listObjectsV2Helper` looks like very much like `listObjectsHelper`, is there anything we can do to DRY this up a bit to reuse common logic?

Reply:
To be honest the refactor is quite doable with templatizing the request object and the actual list objects call but  but I prefer keeping old function as is and just doing switch over.